app/modules/ph/endpoints_ph.py

```diff
@@ -0,0 +1,263 @@
+    return result
+
+
+@module.router.get(
+    "/ph/admin",
+    response_model=list[schemas_ph.PaperComplete],
+    status_code=200,
+)
+async def get_papers_admin(
+    db: AsyncSession = Depends(get_db),
+    user: models_core.CoreUser = Depends(is_user_a_member_of(GroupType.ph)),
+):
+    """
+    Return all editions, sorted from the latest to the oldest
+    """
+    result = await cruds_ph.get_papers(
+        db=db,
+    )  # Return all papers from the latest to the oldest
+    return result
+
+
+@module.router.post(
+    "/ph/",
+    response_model=schemas_ph.PaperComplete,
+    status_code=201,
+)
+async def create_paper(
+    paper: schemas_ph.PaperBase,
+    db: AsyncSession = Depends(get_db),
+    user: models_core.CoreUser = Depends(is_user_a_member_of(GroupType.ph)),
+    notification_tool: NotificationTool = Depends(get_notification_tool),
+):
+    """Create a new paper."""
+
+    paper_complete = schemas_ph.PaperComplete(
+        id=str(uuid.uuid4()),
+        **paper.model_dump(),
+    )
+    try:
+        paper_db = models_ph.Paper(
+            id=paper_complete.id,
+            name=paper_complete.name,
+            release_date=paper_complete.release_date,
+        )
+
+        now = datetime.now(UTC)
+
+        if paper_db.release_date >= now.date() - timedelta(days=30):
+            message = Message(
+                context=f"ph-{paper_db.id}",
+                is_visible=True,
+                title=f"📗 PH - {paper_db.name}",
+                content="A new paper has been released! 🎉",
+                delivery_datetime=datetime.fromisoformat(
+                    paper_db.release_date.isoformat(),
+                ),
+                # The notification will expire in 10 days
+                expire_on=now + timedelta(days=10),
+            )
+            await notification_tool.send_notification_to_topic(
+                custom_topic=CustomTopic(topic=Topic.advert),
+                message=message,
+            )
+        return await cruds_ph.create_paper(paper=paper_db, db=db)
+
+    except ValueError as error:
+        raise HTTPException(
+            status_code=400,
+            detail=str(error),
+        )
+
+
+@module.router.post(
+    "/ph/{paper_id}/pdf",
+    status_code=201,
+)
+async def create_paper_pdf_and_cover(
+    paper_id: str,
+    pdf: UploadFile = File(...),
+    user: models_core.CoreUser = Depends(is_user_a_member_of(GroupType.ph)),
+    request_id: str = Depends(get_request_id),
+    db: AsyncSession = Depends(get_db),
+):
+    paper = await cruds_ph.get_paper_by_id(db=db, paper_id=paper_id)
+    if paper is None:
+        raise HTTPException(
+            status_code=404,
+            detail="The paper does not exist.",
+        )
+
+    await save_file_as_data(
+        upload_file=pdf,
+        directory="ph/pdf",
+        filename=str(paper_id),
+        request_id=request_id,
+        max_file_size=10 * 1024 * 1024,  # 10 MB
+        accepted_content_types=["application/pdf"],
```

Comment:
```suggestion
        accepted_content_types=ContentType.pdf,
```

Reply:
Are you sure ? Because save_file_as_data need a list of string.
https://github.com/aeecleclair/Hyperion/blob/9b6022c825a0203e9aaf693b61ff08ced00454b2/app/utils/tools.py#L107

Reply:
I think the branch is not up-to-date. This should be an enum: https://github.com/aeecleclair/Hyperion/blob/a8a265759ccc2b2af6cc40dcda125ced8ae8c6e1/app/utils/tools.py#L133
I suggest you to merge main into this branch